mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -38,21 +38,22 @@ pub enum SwapWatcherMsg {
 pub struct TakerSwapWatcherData {
     pub uuid: Uuid,
     pub secret_hash: Vec<u8>,
-    pub taker_spends_maker_payment_preimage: Vec<u8>,
-    pub taker_refunds_payment: Vec<u8>,
+    pub maker_payment_spend_preimage: Vec<u8>,
+    pub taker_payment_refund_preimage: Vec<u8>,
     pub swap_started_at: u64,
     pub lock_duration: u64,
     pub taker_coin: String,
     pub taker_fee_hash: Vec<u8>,
     pub taker_payment_hex: Vec<u8>,
-    pub taker_payment_lock: u64,
     pub taker_pub: Vec<u8>,
     pub taker_coin_start_block: u64,
     pub taker_payment_confirmations: u64,
     pub taker_payment_requires_nota: Option<bool>,
     pub taker_amount: BigDecimal,
```

Comment:
After your [comment](https://github.com/KomodoPlatform/atomicDEX-API/pull/1552#discussion_r1028257745), watchers get only the taker payment hash from the watcher message, then retrieve the transaction from the chain. The amount was used to validate if the retrieved transaction is as expected. If we get the taker amount from the transaction, we'd be validating the transaction with itself. So we can either keep it as it is as an extra validation, or just remove the field together with all usages. I think the validation is not that useful because an attacker can just put any value in the field. And if it's not an attacker then everything will be correct anyways.

Reply:
> So we can either keep it as it is as an extra validation, or just remove the field together with all usages. I think the validation is not that useful because an attacker can just put any value in the field. And if it's not an attacker then everything will be correct anyways.

Yeah, and on the watcher side, it might be enough to validate that payment output is actual HTLC that matches the redeem script from `taker_refunds_payment`.